### PR TITLE
[#7369] Can send tokens in Aragon Dapp in Finance section

### LINKED ISSUE #7369: Can send tokens in Aragon Dapp in Finance section
***Type***: Bug

***Summary***: Second `Send transaction` screen does not appear when sending tokens withing Aragon Dapp from it's Finance section (Demo project is used). It's all fine when sending ETH though.

#### Expected behavior
Second transaction appears when sending tokens withing Aragon Dapp from Finance section (Demo project is used)

#### Actual behavior
Second transaction does not appear when sending tokens withing Aragon Dapp in Finance section (Demo project is used). `Waiting for Signature...` screen stays. 
screenshot 2019-01-29 at 14 30 21

 

### Reproduction

- Open Status and recover an account with positive ETH and SNT balance on Mainnet
- Navigate to `+` -> `Open Dapp` -> `Aragon` -> `Open`
- At the Dapp screen bottom tap 'Demo organisation'
- Tap top-left menu button -> `Finance`
- Tap top-right Aragon wallet button to get to `New Transfer` screen
- In `TOKEN` list select SNT and enter valid amount <= your SNT balance
-  Tap 'Create transaction' button
- Sign first transaction from 'Send transaction' screen. Tap 'Done' on confirmation screen.
**Actual result:** Aragon's screen with the `Waiting for Signature...` message displays. No second 'Send transaction' screen appears.

**Expected result:** Second 'Send transaction' screen appears.

- Try to send any other (or again SNT) token from 'New transfer' screen (Top-right wallet button -> Submit deposit -> Create transaction)
**Actual result:** `Waiting for Signature...` screen appears and not Status wallet 'Send transaction' screen appears.

### Additional Information
* Status version: Release 0.9.32, Develop 0.9.33 (2019012902) nightly 
* Operating System: Android and iOS

[logcat.log](https://github.com/status-im/status-react/files/2807697/logs.log) starting from navigating to Aragon Dapp

### PR DESCRIPTION
**Pull request by flexsurfer**
fixes #7369

QA: you can find a button for this case in test Dapp

image

### CODE REVIEW

File: src/status_im/ui/screens/wallet/send/events.cljs [hunk 1]
```diff
@@ -132,7 +132,7 @@
          {:keys [payload message-id] :as queued-transaction} (last transactions-queue)
          {:keys [method params id]} payload
          db' (update-in db [:wallet :transactions-queue] drop-last)]
-     (when (and (not (:id send-transaction)) queued-transaction)
+     (when (and (not (= :wallet-transaction-sent (:view-id db))) (not (:id send-transaction)) queued-transaction)
```
Comment (jeluard) [inline src/status_im/ui/screens/wallet/send/events.cljs:135]: Can you add some comment to detail this?
Is it the proper fix or a workaround?

Comment (goranjovic) [reply src/status_im/ui/screens/wallet/send/events.cljs:135]: same question from me

Comment (flexsurfer) [reply src/status_im/ui/screens/wallet/send/events.cljs:135]: in the current implementation, it looks like a proper fix, so maybe @goranjovic can answer better if we can improve it in wallet refactoring branch

**Comment by flexsurfer** [comment]
@jeluard @goranjovic could you please approve PR ? we want to merge it today to have it in next release tomorrow , thanks

**Review by jeluard — APPROVED** [review]

**Review by goranjovic — APPROVED** [review]

**Comment by Serhy** [comment]
Looks good (iOS and Android 7 tested)
TestDapp both send transaction in batch and one-after-another are working.
With Aragon I can send token which is not in the default list by signing two transactions one-after-another. Can send existing token in Finance as well.
Can sell token in Airswap (two transactions required)
Thanks a lot, @flexsurfer for the fix (and for the test button in TestDapp!)

**Pull request merged**